server/service/src/sync/api_v6/mod.rs

```diff
@@ -0,0 +1,113 @@
+mod core;
+
+use repository::RepositoryError;
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use util::format_error;
+
+use crate::i64_to_u64;
+
+pub use self::core::*;
+
+use super::{
+    api::{
+        CommonSyncRecord, ParsedError, ParsingResponseError, SyncApiError, SyncApiErrorVariantV5,
+    },
+    translations::PushSyncRecord,
+    SyncApiSettings,
+};
+
+#[derive(Deserialize, Debug, Error, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SyncParsedErrorV6 {
+    #[error("Problem connecting to legacy server: {0}")]
+    CannotConnectToLegacyServer(String),
+    #[error(transparent)]
+    LegacyServerError(#[from] ParsedError),
+    #[error("Other legacy server error: {0}")]
+    OtherLegacyServerError(String),
+    #[error("Other server error: {0}")]
+    OtherServerError(String),
+    #[error("Not a central server")]
+    NotACentralServer,
+}
+
+impl From<SyncApiError> for SyncParsedErrorV6 {
+    fn from(from: SyncApiError) -> Self {
+        use SyncApiErrorVariantV5 as FromError;
+        use SyncParsedErrorV6 as ToError;
+
+        let formated_error = format_error(&from);
+        match from.source {
+            FromError::ParsedError { source, .. } => ToError::LegacyServerError(source),
+            FromError::ConnectionError(_) => ToError::CannotConnectToLegacyServer(formated_error),
+            _ => ToError::OtherLegacyServerError(formated_error),
+        }
+    }
+}
+
+impl From<RepositoryError> for SyncParsedErrorV6 {
+    fn from(from: RepositoryError) -> Self {
+        SyncParsedErrorV6::OtherServerError(format_error(&from))
+    }
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SyncPullResponseV6 {
+    Data(SyncBatchV6),
+    Error(SyncParsedErrorV6),
+}
+
+#[derive(Error, Debug)]
+#[error("Sync api error, url: '{url}', route: '{route}'")]
+pub(crate) struct SyncApiErrorV6 {
+    pub source: SyncApiErrorVariantV6,
+    pub(crate) url: Url,
+    pub(crate) route: String,
+}
+
+#[derive(Error, Debug)]
+pub(crate) enum SyncApiErrorVariantV6 {
+    #[error("Connection problem")]
+    ConnectionError(#[from] reqwest::Error),
+    #[error("Could not parse respose")]
+    ParsedError(#[from] SyncParsedErrorV6),
+    #[error("Could not parse respose")]
+    ParsingResponseError(#[from] ParsingResponseError),
```

Comment:
> There is a spelling error in response

a64540d2926549d8c4586fa380fced819b0fe2c1, thanks

> The variant naming is inconsistent "Parsed" vs "Parsing"...

I thought they are different, since 'parsing' error is before it's parsed, and parsed error indicates we matched an error from api response. Not 100% happy with those variants, but not sure what's the best alternative